Core/clim-core/builtin-commands.lisp

```diff
@@ -428,3 +428,36 @@
         (funcall *sys-read-preserving-whitespace*
                  stream eof-error-p eof-value recursivep)))
 ) ; with-system-redefinition-allowed
+
+;;;
+;;;  Copy to clipboard is a global command that should be available
+;;;  everywhere, just like the Describe command above.
+;;;
+(defclass clipboard-object ()
+  ((content :initarg :content
+            :reader clipboard-object/content)
+   (type    :initarg :type
+            :reader clipboard-object/type)))
+
+(clim:define-presentation-translator presentation-clipboard-object
+    (t clipboard-object clim:global-command-table :tester ((obj presentation)
+                                                           (supported-clipboard-types obj (presentation-type presentation))))
+    (obj presentation)
+  (make-instance 'clipboard-object :content obj :type (presentation-type presentation)))
+
+(clim:define-command (com-copy-to-clipboard :command-table clim:global-command-table :name "Copy Presentation to Clipboard")
+    ((obj clipboard-object :prompt "Object"))
+  (let ((pane (clim:frame-top-level-sheet clim:*application-frame*)))
+    (copy-to-clipboard (clim:port pane) pane (clipboard-object/content obj)
+                       :presentation-type (clipboard-object/type obj))))
```

Comment:
I can see that I can "Copy Presentation to Clipboard" disregarding its type (i.e forms in the interactor), but nothing happens to be in the clipboard when there is no translator. Is there a way to make the command abort, fail, or refuse to accept presentations which doesn't have appropriate translators?

Steps to reproduce:
run stream-test
issue any command
run command `copy presentation to clipboard` and select previous form
try to paste (nothing happens)


Second issue: copy text command if there is no text selected should not complete successfully either.

Reply:
I've added the commands.

Reply:
For the failure if nothing is selected, how should the failure manifest itself? It's definitely possible to print a message to `*STANDARD-OUTPUT*`, but that might not always be preferable. The information is there, it's just a matter of how to report it back to the user.

The other issue is a trickier one. When you selected the previous form, you're actually not selecting it. Rather, you are reissuing the previous command. The way to show this is the case is to do the following:

  - Type `Clear Paste Pane` on the commandline
  - Paste some text info the paste pane
  - Type `Copy Presentation to Clipboard` followed by space so that the commandline prompts for a presentation.
  - Click on the `Clear Paste Pane` command that was previously typed

After this, the paste pane is cleared. What happens is that it looks like the `Clear Paste Pane` command was chosen, but it was in fact executed instead.

I believe this is a bug that is unrelated to the clipboard feature. It shouldn't even be possible to select the previous command at this point, since its presentation does not have a converter (one can, and probably should, be written, but that's s separate issue).

Reply:
you may trigger "command aborted". that will prevent pane being redisplayed(!)

as of the issue, please report it with steps to reproduce then.

Reply:
That sounds like a good idea. Only problem is that I don't know how to trigger a command aborted. I'll experiment.

I will report the issue separately with a simple test case. I should be able to create one that doesn't require copying and pasting stuff.

Reply:
I have now updated the copy command so that it triggers an abort when the selection is not active.

Fixed in 77cb2feaf6e370dcd7ed33d41193c22f02105dfa